src/Spice86.Core/Emulator/Memory/Memory.cs

```diff
@@ -9,7 +9,7 @@ public class Memory {
     private readonly IMemoryDevice _ram;
     private readonly BreakPointHolder _readBreakPoints = new();
     private readonly BreakPointHolder _writeBreakPoints = new();
-    private IMemoryDevice[] _memoryDevices;
+    private List<IMemoryDevice> _memoryDevices;
```

Comment:
Why is this? 
It's not supposed to be searched, sorted or manipulated. It's a straight-up mapping between a memory address and the class that provides access to that memory.